get_checks.py

```diff
@@ -0,0 +1,189 @@
+import os
+import re
+import yaml
+from yaml.representer import SafeRepresenter
+from Ska.Shell import bash
+from Ska.File import chdir
+
+# If content in these fields is present in the input yaml, it should be preserved
+SAVE_FIELDS = ['title', 'aca_cl_id', 'notes', 'hopper', 'ignore', 'type', 'severity', 'byhand', 'missing']
+
+
+def get_options():
+    import argparse
+    parser = argparse.ArgumentParser(
+        description="Get starcheck checks")
+    parser.set_defaults()
+    parser.add_argument("scdir",
+                        help="starcheck git repo directory")
+    parser.add_argument("--infile",
+                        default="checks.yaml",
+                        help="file of checks (output from previous run of tool)")
+    parser.add_argument("--outfile",
+                        default="checks.yaml")
+    args = parser.parse_args()
+    return args
+
+# Add helper code to be able to print certain fields (context) as literal blocks
+
+
+class literal_str(str):
+    pass
+
+
+def change_style(style, representer):
+    def new_representer(dumper, data):
+        scalar = representer(dumper, data)
+        scalar.style = style
+        return scalar
+    return new_representer
+
+represent_literal_str = change_style('|', SafeRepresenter.represent_str)
+yaml.add_representer(literal_str, represent_literal_str)
+
+
+def get_dir_version(dirname):
+    """
+    Get git revision of the supplied repo dir
+    Print a warning if the try looks dirty.
+    """
+    with chdir(dirname):
+        changes = int(bash("git diff-index --quiet HEAD ; echo $?")[0])
+        if changes != 0:
+            print "Warning: {} has changes and git version may not map to code".format(dirname)
+        return bash("git rev-parse HEAD")[0]
+
+
+def get_single_warn(fname, ftext, idx, context_range=[-15, 10]):
+    """
+    Extra the context and text of a warning
+
+    :param fname: file name with code and warning lines
+    :param ftext: the text lines of that file
+    :param idx: line index of a warning
+    :param context_range: lines back and forward of idx to fetch for context
+    :return: dictionary of the warning
+    """
+    warn_text = []
+    for fline in ftext[idx:]:
+        warn_text.append(fline.strip())
+        if re.search(";", fline):
+            break
+    min_line = max(idx + context_range[0], 0)
+    max_line = min(idx + context_range[1], len(ftext) - 1)
```

Comment:
And I think I need to add 1 to the context_range[1] to get a full ten lines after the line that includes the warning.